File: src/main/java/pl/SFG/SGF/dto/fighting/FighterStatsDto.java
package pl.SFG.SGF.dto.fighting;

import lombok.Builder;
import lombok.Getter;
import lombok.Setter;
import pl.SFG.SGF.dto.hero.HeroClass;

@Getter
@Builder
@Setter
public class FighterStatsDto {

    private Long id;

    private String name;

    private String avatarUrl;

    private HeroClass heroClass;
    private int health;

    private int attack;

    private int magic;

    private float speed;

    private int shield;

    private int luck;
}

File: frontend/SFG_Frontend/src/features/fighting/FighterStatsDto.ts
import { HeroClass } from "./HeroClass";

export interface FighterStatsDto {
  id: number;
  name: string;
  avatarUrl?: string;
  heroClass: HeroClass;

  health: number;
  attack: number;
  magic: number;
  speed: number;
  shield: number;
  luck: number;
}